### PR TITLE
feat: add AMMV-aware social force diagnostics

### LINKED ISSUE #2154: model: add AMMV-aware social-force interaction term and diagnostics
## Goal / Problem

Add a configurable AMMV-aware social-force interaction term and diagnostics for pedestrian response to the vehicle.

The first implementation should be explicitly diagnostic unless calibrated pedestrian/AMMV interaction data is available.

## Hypothesis

An AMMV-specific interaction term that accounts for footprint, speed, heading, relative bearing, time-to-collision, lateral clearance, swept path, and visibility state will produce inspectable pedestrian trajectory differences compared with classical SFM on matched scenarios.

Evidence grade: speculative.

## Scope and non-goals

In scope:
- Add a versioned parameter contract for AMMV-aware social-force behavior.
- Make the AMMV term configurable so classical SFM and AMMV-aware SFM can run on identical scenarios.
- Add diagnostics such as pedestrian lateral deviation, speed adaptation, minimum pedestrian-AMMV clearance, interaction duration, force magnitude over time, personal-space intrusion count, and group disruption proxy where available.
- Add unit tests for force direction, zero-force limits, monotonic distance behavior, and bounded outputs.
- Produce one small diagnostic report showing at least one scenario where the AMMV term changes pedestrian trajectories.

Non-goals:
- Claiming calibrated human response without source data.
- Replacing existing SFM behavior globally.
- Treating diagnostic trajectory changes as benchmark improvement.
- Adding broad new planner families.

## Acceptance criteria

- [ ] AMMV-aware force behavior can be enabled and disabled by config.
- [ ] Classical SFM and AMMV-aware SFM can be compared on identical scenarios.
- [ ] Unit tests cover direction, zero-force limits, monotonic distance behavior, and bounded outputs.
- [ ] A compact diagnostic report records a trajectory difference and states the claim boundary.
- [ ] The research-v1 evidence map is updated or explicitly marked unchanged.

## Research result guidance

Target claim or hypothesis: AMMV-aware interaction modeling can be represented and inspected within Robot SF.

Comparator or baseline: classical SFM with the AMMV term disabled on the same scenario/seed.

Minimum valid evidence: passing unit tests plus one reproducible diagnostic scenario comparison.

Decision or stop rule: stop at diagnostic model evidence unless calibrated source data is staged and accepted.

Expected artifacts: config, tests, compact report, and optional small tracked evidence summary.

Parent issue / claim map / context note to update: research-v1 evidence map and any SFM/modeling context note created for this work.

## Validation command

Expected targeted validation, adjusted to touched paths:

```bash
uv run pytest tests -k "social_force or ammv" -q
uv run ruff check <touched-python-files>
uv run ruff format --check <touched-python-files>
```

## Artifact policy

Evidence category: diagnostic model evidence.

`output/` usage: local generated traces may be used during iteration but must not be the durable record.

Durable pointer or tracked evidence: compact tracked report/summary with command, config, seed, and checksum if generated artifacts support the conclusion.

## Estimate metadata

Effort (h): 16-32.

Complexity: high; touches model behavior, config contracts, diagnostics, and tests.

Risk: high; easy to overclaim without calibration.

Uncertainty: 40%; effort depends on fast-pysf and Robot SF integration boundaries.

### PR DESCRIPTION
**Pull request by ll7**
## Summary
- Closes #2154.
- Adds an optional AMMV-aware Social Force interaction term, disabled by default, with diagnostics exposed through planner metadata.
- Adds a diagnostic AMMV baseline config, targeted tests, and a context note/evidence-map update marking the result as diagnostic only.

## Validation
- `uv run pytest tests/baselines/test_social_force.py -q`
- `uv run pytest tests -k 'social_force or ammv' -q`\n- `uv run ruff check robot_sf/baselines/social_force.py tests/baselines/test_social_force.py`\n- `uv run ruff format --check robot_sf/baselines/social_force.py tests/baselines/test_social_force.py`\n- `uv run python -m py_compile robot_sf/baselines/social_force.py tests/baselines/test_social_force.py`\n- `BASE_REF=origin/main scripts/dev/check_docs_proof_consistency_diff.sh`\n- `BASE_REF=origin/main scripts/dev/pr_ready_check.sh` produced `status: passed` for `52a4f48f159fe3627b3581923dabd24afb553feb`; wrapper was terminated after writing the success stamp because an orphan worker kept the shell open.\n\n## Research Result Guidance\n- Target claim/hypothesis/blocker: `research-v1.amv.model_behavior`, AMMV-aware Social Force can be probed as a named mechanism without breaking existing planner contracts.\n- Comparator/baseline: default Social Force behavior remains unchanged when AMMV is disabled.\n- Evidence tier: diagnostic mechanism tests and config contract only.\n- Result classification: diagnostic, not benchmark-strength and not paper-facing.\n- Decision/stop rule: keep claim below candidate until a paired classical Social Force vs AMMV-aware run on a named AMMV scenario/config/seed has durable summary provenance.\n- Synthesis surface/update target: `docs/context/issue_2153_research_v1_evidence_map.md` and `docs/context/issue_2154_ammv_social_force_model.md`.\n\n## Downstream Propagation\n- Parent issue/claim map: updated `docs/context/issue_2153_research_v1_evidence_map.md`.\n- Context index/catalog: updated `docs/context/README.md`, `docs/context/INDEX.md`, and `docs/context/catalog.yaml`.\n- Leaderboard/benchmark report: NA; no benchmark campaign or performance claim.\n- Artifact catalog/registry: NA; no generated `output/` artifact is required for durable evidence.\n- Follow-up: paired diagnostic run should be a separate issue or child PR before claim movement.\n

## Release Notes

* **New Features**
  * Added AMMV-aware Social Force baseline with optional speed-scaled pedestrian repulsion
  * Includes configurable repulsion amplitude, range, and speed scaling parameters

* **Documentation**
  * Added detailed configuration reference and setup guide for AMMV-aware Social Force
  * Includes diagnostic and validation instructions

### CODE REVIEW

**Comment by ll7** [comment]
Follow-up proof step created as #2168 for the paired classical Social Force vs AMMV-aware diagnostic. PR #2167 intentionally stops at diagnostic mechanism evidence; no benchmark-strength or paper-facing claim is made here.

File: robot_sf/baselines/social_force.py [hunk 1]
```diff
@@ -398,6 +410,150 @@ def _compute_total_force(
...
+
+        assert self._wrapper is not None
+
+        ped_pos = self._wrapper.sim.peds.pos()
+        n_peds = ped_pos.shape[0]
+        if n_peds == 0:
+            self._last_ammv_diagnostics = diagnostics
+            ammv_force = np.zeros(2, dtype=float)
+            self._store_ammv_diagnostics(ammv_force, diagnostics)
+            return ammv_force
```
[the remark this answers is not in the document]
Comment (ll7) [reply robot_sf/baselines/social_force.py]: Fixed in 61524304: AMMV now builds pedestrian position/velocity arrays from the current observation and prefetches ped_vels once before the loop.

File: robot_sf/baselines/social_force.py [hunk 2]
```diff
@@ -398,6 +410,150 @@ def _compute_total_force(
...
+                else np.zeros(2, dtype=float)
+            )
+            relative_pos = ped_pos[i] - robot_pos
+            relative_vel = ped_vel - robot_vel
+            lateral_clearance = abs(
+                float(robot_vel[0] * relative_pos[1] - robot_vel[1] * relative_pos[0])
+            ) / max(
+                robot_speed,
+                self.EPSILON,
+            )
```
[the remark this answers is not in the document]
Comment (ll7) [reply robot_sf/baselines/social_force.py:510]: Fixed in 61524304: stationary robots now use a stable reference direction for lateral clearance, and pedestrian velocities are pre-fetched. I did not cache config values in __init__ because this planner supports runtime configure() updates.

File: robot_sf/baselines/social_force.py [hunk 3]
```diff
@@ -398,6 +410,150 @@ def _compute_total_force(
...
+            magnitude = (
+                self.config.ammv_repulsion_amplitude * speed_mult * np.exp(-dist / effective_range)
+            )
+            total += direction * magnitude
+
+            ped_vel = (
+                np.asarray(self._wrapper.sim.peds.vel()[i], dtype=float)
+                if hasattr(self._wrapper.sim.peds, "vel")
+                else np.zeros(2, dtype=float)
+            )
```
[the remark this answers is not in the document]
Comment (ll7) [reply robot_sf/baselines/social_force.py]: Fixed in 61524304: _compute_total_force() now passes live obs.agents into the AMMV helper, so force and diagnostics use the current observation instead of the initialized simulator pedestrian state.

File: tests/baselines/test_social_force.py [hunk 4]
```diff
@@ -349,5 +401,216 @@ def test_social_force_planner_accepts_dict_observation_without_obstacles(self):
...
+                "goal": [2.0, 0.0],
+                "radius": 0.3,
+            },
+            agents=[{"position": [0.5, 0.0], "velocity": [0.0, 0.0], "radius": 0.3}],
+            obstacles=[],
+        )
+        act_base = planner_base.step(obs)
+        act_ammv = planner_ammv.step(obs)
+        # AMMV with large amplitude should produce measurably different vy deflection
+        assert abs(act_ammv["vy"] - act_base["vy"]) > 0.001
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/baselines/test_social_force.py]: Fixed in 61524304: the enabled-AMMV test now asserts the x-component changes for the pedestrian geometry in that test.

File: tests/baselines/test_social_force.py [hunk 5]
```diff
@@ -349,5 +401,216 @@ def test_social_force_planner_accepts_dict_observation_without_obstacles(self):
...
+            agents=[{"position": ped_pos, "velocity": [0.0, 0.0], "radius": 0.3}],
+            obstacles=[],
+        )
+        _ = planner_fast.step(obs2)
+        meta_fast = planner_fast.get_metadata()
+        # Slow planner: stationary velocity
+        _ = planner_slow.step(obs)
+        meta_slow = planner_slow.get_metadata()
+        # Fast-moving robot should have larger AMMV force magnitude
+        assert meta_fast["ammv_force_magnitude"] >= meta_slow["ammv_force_magnitude"] * 0.9
```
[the remark this answers is not in the document]
Comment (ll7) [reply tests/baselines/test_social_force.py]: Fixed in 61524304: the speed-scaling test now uses identical robot/pedestrian geometry and varies only robot velocity.
